test/e2e/storage/persistent_volumes-disruptive.go

```diff
@@ -218,22 +240,48 @@ func initTestCase(f *framework.Framework, c clientset.Interface, pvConfig framew
-	framework.ExpectNoError(framework.DeletePersistentVolume(c, pv.Name), "tearDown: Failed to delete PV ", pv.Name)
+func tearDownTestCase(c clientset.Interface, f *framework.Framework, ns string, client, server *v1.Pod, pvc *v1.PersistentVolumeClaim, pv *v1.PersistentVolume) {
+	framework.DeletePodWithWait(f, c, client)
+	framework.DeletePersistentVolumeClaim(c, pvc.Name, ns)
+	framework.DeletePersistentVolume(c, pv.Name)
+	framework.DeletePodWithWait(f, c, server)
 }
 
-// kubeletCommand performs `start`, `restart`, or `stop` on the kubelet running on the node of the target pod.
+// kubeletCommand performs `start`, `restart`, or `stop` on the kubelet running on the node of the target pod and waits
+// for the desired statues..
+// - First issues the command via `systemctl`
+// - If `systemctl` returns stderr "command not found, issues the command via `service`
+// - If `service` also returns stderr "command not found", the test is aborted.
 // Allowed kubeletOps are `kStart`, `kStop`, and `kRestart`
+// TODO return error instead of handling internally
 func kubeletCommand(kOp kubeletOpt, c clientset.Interface, pod *v1.Pod) {
+	systemctlCmd := fmt.Sprintf("sudo systemctl %s kubelet", string(kOp))
+	serviceCmd := fmt.Sprintf("sudo service kubelet %s", string(kOp))
 	nodeIP, err := framework.GetHostExternalAddress(c, pod)
 	Expect(err).NotTo(HaveOccurred())
 	nodeIP = nodeIP + ":22"
-	sshResult, err := framework.SSH("sudo /etc/init.d/kubelet "+string(kOp), nodeIP, framework.TestContext.Provider)
-	Expect(err).NotTo(HaveOccurred())
-	framework.LogSSHResult(sshResult)
 
+	framework.Logf("Attempting `%s`", systemctlCmd)
+	sshResult, err := framework.SSH(systemctlCmd, nodeIP, framework.TestContext.Provider)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("SSH to Node %q errored.", pod.Spec.NodeName))
+	framework.LogSSHResult(sshResult)
+	if strings.Contains(sshResult.Stderr, "command not found") /*"command not found"*/ {
+		framework.Logf("Command `%s` returned code <%d>: %q.", systemctlCmd, sshResult.Code, sshResult.Stderr)
+		framework.Logf("Attempting %s", serviceCmd)
+		sshResult, err = framework.SSH(serviceCmd, nodeIP, framework.TestContext.Provider)
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("SSH to Node %q errored.", pod.Spec.NodeName))
+		framework.LogSSHResult(sshResult)
+		if strings.Contains(sshResult.Stderr, "command not found") /*"command not found"*/ {
+			framework.Logf("Command `%s` returned code <%d>: %q.", serviceCmd, sshResult.Code, sshResult.Stderr)
+			Expect(sshResult.Code).To(BeZero(), fmt.Sprintf("Unable to [%s] kubelet.", string(kOp)))
+		} else if sshResult.Code != 0 {
+			// Error other than "command not found"
+			Expect(err).To(BeZero(), "Failed to [%s] kubelet:\n%#v", string(kOp), sshResult)
+		}
+	} else if sshResult.Code != 0 {
+		// Error other than "command not found"
+		Expect(err).To(BeZero(), "Failed to [%s] kubelet:\n%#v", string(kOp), sshResult)
```

Comment:
Should it be either Expect(err).NotTo(HaveOccurred()) or Expect(sshResult.Code).To(BeZero())?

 I think it is possible that sshResult.Code is 0 but err != nil. This means that SSH failed.
The old code only checks err is nil or not.